docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+```
+data:
+  external.metrics.default.nginxext-nginx.net.request_per_s: '{"name":"nginx.net.request_per_s","labels":{"app":"puppet","env":"demo"},"ts":1530625976,"hpa_name":"nginxext","hpa_namespace":"default","value":2493,"valid":true}'
+```
+In the ConfigMap, indicating that it works.
+
+- If you see:
+```
+Conditions:
+  Type           Status  Reason                   Message
+  ----           ------  ------                   -------
+  AbleToScale    True    SucceededGetScale        the HPA controller was able to get the target's current scale
+  ScalingActive  False   FailedGetExternalMetric  the HPA was unable to compute the replica count: unable to get external metric default/nginx.net.request_per_s/&LabelSelector{MatchLabels:map[string]string{kube_container_name: nginx,},MatchExpressions:[],}: unable to fetch metrics from external metrics API: the server could not find the requested resource (get nginx.net.request_per_s.external.metrics.k8s.io)
+
+```
+
+Then it's likely that you don't have the proper RBAC set for the HPA.
+Make sure that `kubectl api-versions` shows:
+```
+autoscaling/v2beta1
+[...]
+external.metrics.k8s.io/v1beta1 
+```
+The latter will show up if the DCA properly registers as an External Metrics Provider.
+
+And that you have the same service name referenced in the APIService for the External Metrics Provider and the one for the DCA serving on port 443.
+Also make sure you have created the RBAC from the Register the External Metrics Provider step.
+
+- If you see
+
+```
+  Warning  FailedComputeMetricsReplicas  3s (x2 over 33s)  horizontal-pod-autoscaler  failed to get nginx.net.request_per_s external metric: unable to get external metric default/nginx.net.request_per_s/&LabelSelector{MatchLabels:map[string]string{kube_container_name: nginx,},MatchExpressions:[],}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get nginx.net.request_per_s.external.metrics.k8s.io)
+```
+
+Make sure the DCA is running and the service exposing the port 443 and which name is registered in the APIService are up.
+
+- You are not collecting the service tag from the DCA
+Make sure the service map is available by exec'ing into the DCA pod and run:
+`datadog-cluster-agent metamap`
+Then, make sure you have the same secret (or a 32 characters long) token referenced in the agent and in the DCA.
+The best is to check in the environment variables (just type `env` when in the agent or the DCA pod).
+Then make sure you have the `DD_CLUSTER_AGENT` option turned on in the node agent's manifest.
+
+- Why am I not seeing the same value in Datadog and in Kubernetes ?
+
+As Kubernetes autoscales your resources the current target will be weighted by the number of replicas of the scaled deployement.
+So the value returned by the DCA is fetched from Datadog should be proportionnaly equal to the current target times the number of replicas. 
```

Comment:
r/`So the value returned by the DCA is fetched from Datadog should be proportionnaly equal to the current target times the number of replicas. ` / `So the value returned by the DCA is fetched from Datadog and should be proportionally equal to the current target times the number of replicas. `